### ISSUE TITLE
Unable to locate OBD and OOD modules in the repository

### ISSUE THREAD

**Issue by HHuiwen** [LONGXUANX/CDFormer_code#2]
Hi, thank you for your excellent work!
I was wondering if you could kindly point me to where the OBD and OOD modules are implemented in the codebase?

**Comment by HHuiwen** [comment]
> In the paper, they are named as modules (mainly because the teacher asked to name them so), but actually I would like to name them as two strategies. So these two are not specific modules, but operations within the entire network. For instance, OBD is the strategy when calculating similarity, and OOD is the strategy when calculating loss. Besides, the focus of our work is on the design of the network structure and the support of background placeholders for arbitrary class inference. OOD and OBD are embedded within it.

Thank you very much for your reply. I really appreciate the elegance of your training strategy—it provides a seamless integration of the OOD and OBD concepts into the overall network design. 

Could you kindly point me to where the feature space $T = \{t_1, t_2, \ldots, t_C\}$ used in the OOD strategy is defined and first utilized, as well as where the background placeholder $\texttt{[bg]}$ and the learnable token $T_b$ used in the OBD strategy are defined and initially applied? :)

**Comment by HHuiwen** [comment]
> By using the POWER OF GOD, I discovered that the learnable background feature is defined in attention.py under the class SingleHeadSiameseAttention(nn.Module) at the line: self.dummy = nn.Parameter(torch.Tensor(1, self.d_model)) And the background placeholder is defined as: dummy_v = torch.zeros(sz_b, 1, self.n_head, self.d_model, device=v.device)

Thanks, it helps a lot!!

**Comment by gitfrahts** [comment]
感谢你的提问; 感谢你的回答;好人一生平安

> > 通过使用 POWER OF GOD，我发现可学习的背景功能在 attention.py 类 SingleHeadSiameseAttention（nn.Module） 在以下行中： self.dummy = nn.参数（torch.Tensor（1， self.d_model）） 背景占位符定义为： dummy_v = torch.zeros（sz_b， 1， self.n_head， self.d_model， device=v.device）
> 
> 谢谢，它帮了大忙！！



> 通过使用 POWER OF GOD，我发现可学习的背景功能在 attention.py 类 SingleHeadSiameseAttention（nn.Module） 在以下行中： self.dummy = nn.参数（torch.Tensor（1， self.d_model）） 背景占位符定义为： dummy_v = torch.zeros（sz_b， 1， self.n_head， self.d_model， device=v.device）
